### PR TITLE
Update content in sign-in/up pages and emails

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

We currently don't have parity with the content in the prototype relating to signing up and signing in.

### Changes proposed in this pull request

This PR makes a number of small content changes, see per commit.

### Guidance to review

- Double check that content has been correctly updated.
- The "Check your email" pages for signing up and signing in are identical according to the prototype so I've made it a shared view. Is this a premature refactoring?

### Link to Trello card

[110 - Improve content in sign-in/up pages and emails](https://trello.com/c/LH9cajy7/110-improve-content-in-sign-in-up-pages-and-emails)

### CODE REVIEW

File: app/views/authentication_mailer/sign_in_email.text.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # <%= t('authentication.sign_in.email.heading') %>
```
Comment (alexdesi) [inline app/views/authentication_mailer/sign_in_email.text.erb]: not sure if we need this comments

File: app/controllers/candidate_interface/sign_up_controller.rb [hunk 2]
```diff
@@ -9,10 +9,10 @@ def create
...
         MagicLinkSignIn.call(candidate: @candidate)
-        render 'candidate_interface/sign_in/show'
+        render 'candidate_interface/shared/check_your_email'
       elsif @candidate.save
         MagicLinkSignUp.call(candidate: @candidate)
-        render :show
+        render 'candidate_interface/shared/check_your_email'
       else
         render :new
       end
```
Comment (alexdesi) [inline app/controllers/candidate_interface/sign_up_controller.rb]: As discussed this is good enough, just have a look at this implementation (not need to use it! I've just had this idea that I wanted to share with you :) )

```Ruby
  class SignUpController < CandidateInterfaceController
    def new
      @candidate = Candidate.new
    end

    def create
      @candidate = Candidate.find_or_initialize_by(candidate_params)

      if @candidate.valid?
        magic_link_email.call(candidate: @candidate)
        render 'candidate_interface/shared/check_your_email'
      else
        render :new
      end
    end

    def show; end

  private

    def candidate_params
      params.require(:candidate).permit(:email_address)
    end

    def magic_link_email
      @candidate.new_record? ? MagicLinkSignUp : MagicLinkSignIn
    end
  end
```

File: app/views/authentication_mailer/sign_in_email.text.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
 
-<%= t('authentication.sign_in.email.body') %>
+Click the link below to sign in to the Apply for teacher training service.
```
Comment (alexdesi) [inline app/views/authentication_mailer/sign_in_email.text.erb]: 💯  I'm glad we are just using plain text for the emails.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -76,6 +76,7 @@
         </span>
         </p>
       </div>
+      <%= yield(:before_content) %>
```
Comment (alexdesi) [inline app/views/layouts/application.html.erb]: maybe we can be a bit more semantic here, which kind of content `before_content` is going to contain? this question should help to name it (in theory :P )

File: config/locales/authentication.yml [hunk 5]
```diff
@@ -9,13 +9,11 @@ en:
...
-        body: Click the link below to sign in to the Apply for postgraduate teacher training service.
     check_your_email: Check your email
     sign_in:
       heading: Sign In
       email:
-        subject: Sign In
-        heading: Please confirm you want to Sign In
-        body: Click the link below to sign in to the Apply for postgraduate teacher training service.
+        subject: Sign in to Apply for teacher training
+        heading: Sign in to Apply for teacher training
```
Comment (alexdesi) [inline config/locales/authentication.yml]: are we using heading?

File: spec/system/candidate_interface/candidate_signing_in_spec.rb [hunk 6]
```diff
@@ -52,4 +52,12 @@
       expect(page).to have_content('Welcome')
     end
   end
+
+  it 'can take the user back to the start page' do
```
Comment (alexdesi) [inline spec/system/candidate_interface/candidate_signing_in_spec.rb]: Let's just Use the third person in the present tense (from betterspec.org)
Screenshot 2019-10-04 at 10 48 22

Comment (alexdesi) [reply spec/system/candidate_interface/candidate_signing_in_spec.rb]: `takes the user back to the start page`

File: spec/system/candidate_interface/candidate_signing_up_spec.rb [hunk 7]
```diff
@@ -78,7 +78,15 @@
     it 'sees the start page' do
       visit candidate_interface_welcome_path(token: 'meow')
 
-      expect(page.current_url).to eq(candidate_interface_start_url)
+      expect(page).to have_current_path(candidate_interface_start_path)
     end
   end
+
+  it 'can take the user back to the start page' do
```
Comment (alexdesi) [inline spec/system/candidate_interface/candidate_signing_up_spec.rb]: as above

File: app/views/authentication_mailer/sign_in_email.text.erb (hunk 1 above)
Comment (tvararu) [reply app/views/authentication_mailer/sign_in_email.text.erb]: This is an .erb file, so this is not a comment, it's a heading:

Screenshot 2019-10-04 at 10 58 14

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (tvararu) [reply app/views/layouts/application.html.erb]: I think `before_content` is fine, it hasn't caused confusion in `manage-courses-frontend`. It's a pretty standard Railsy thing.

Comment (chubberlisk) [reply app/views/layouts/application.html.erb]: Could be a back link or breadcrumbs as they go above the `govuk-main-wrapper`. I stole the name from Find. 😈

What would you suggest?

File: config/locales/authentication.yml (hunk 5 above)
Comment (chubberlisk) [reply config/locales/authentication.yml]: I've updated this to remove `heading` and use the `subject` translation string in the email for the heading of it.

File: spec/system/candidate_interface/candidate_signing_in_spec.rb (hunk 6 above)
Comment (chubberlisk) [reply spec/system/candidate_interface/candidate_signing_in_spec.rb]: Makes sense, I've updated now!

File: spec/system/candidate_interface/candidate_signing_up_spec.rb (hunk 7 above)
Comment (chubberlisk) [reply spec/system/candidate_interface/candidate_signing_up_spec.rb]: See https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/259#discussion_r331444408.

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (alexdesi) [reply app/views/layouts/application.html.erb]: I like `breadcrumbs`, it sounds clearer for me, is it ok for you @tvararu? happy to leave as it is you have a strong opinion

File: app/controllers/candidate_interface/sign_up_controller.rb (hunk 2 above)
Comment (chubberlisk) [reply app/controllers/candidate_interface/sign_up_controller.rb]: Thank you for your suggestion! I can see why you would want to refactor this as we have the `render 'candidate_interface/shared/check_your_email'` duplicated. However I think the current implementation is fine for now, I kind of like how I can read through it in one go i.e. don't need read through and then refer to a private method and think about what the ternary does and then go back to the create method.

**Pull request merged**